src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -74,6 +75,47 @@ UT_STATIC void sys_comp_module_##adapter##_init(void) \
 \
 DECLARE_MODULE(sys_comp_module_##adapter##_init)
 
+#define DECLARE_MODULE_ADAPTER_STEP1(adapter, uuid, tr) \
```

Comment:
sorry, could you explain why this is needed? Why the current `DECLARE_MODULE_ADAPTER()` isn't enough?

Reply:
It's described in commit message, did you notice it. I can't explain better.